### DISCUSSION TITLE
KafkaTopic issue

### DISCUSSION THREAD

**Discussion by shreyasarani23** [strimzi/strimzi-kafka-operator#4923] [General]
I have created a topic called my-topic with the following configuration
apiVersion: kafka.strimzi.io/v1beta2
kind: KafkaTopic
metadata:
  name: my-topic
spec:
  partitions: 3
  replicas: 3
  config:
    min.insync.replicas: 2
    retention.ms: 7200000
    segment.bytes: 1073741824

But when I do Kubectl get KafkaTopic it shows me:

Name                   Cluster                    Partitions  Replication factor Ready
my-topic              my-kafka-cluster       1                      1                   True

but why it is showing paritions and replication factor as 1?
In yaml I have specified it has 3.

**Comment by scholzj** [comment]
Did you disabled topic auto-creation in Kafka? If not, it was probably auto-created by your application first when it connected?

**Comment by shreyasarani23** [comment]
yeah I had enabled auto-creation of topic as true. So I guess my kafkatopic was overwritten by the broker config.
Can we avoid using the topic operator if we decide to create topics by directly accessing the cluster. if we don't use topic operator will it cause any problem?
Then how will be the process of  restoring data from persistent volumes if we decide not to use topic operator??

**Reply by scholzj** [reply]
Yeah, kind of ... it is more like a race condition and in your case it looks like the topic auto-created in Kafka won.

You do not need to use the Topic Operator if you don't want to. You can manage your topics directly in Kafka.
